Makefile

```diff
@@ -276,12 +280,29 @@ unit_tests         := $(wildcard tests/unit/*.tzt)
 unit_tests_failing := $(shell cat tests/failing.unit)
 unit_tests_passing := $(filter-out $(unit_tests_failing), $(unit_tests))
 
-test-unit:         $(unit_tests_passing:=.run)
-test-unit-failing: $(unit_tests_failing:=.run)
+test-unit:         $(unit_tests_passing:=.unit)
+test-unit-failing: $(unit_tests_failing:=.unit)
 
-tests/%.run: tests/% $(llvm_kompiled)
+tests/%.unit: tests/% $(llvm_kompiled)
 	$(TEST) interpret --backend llvm $< --output-file /dev/null
 
+# symbolic
+
+symbolic_tests         := $(wildcard tests/symbolic/*.tzt)
+symbolic_tests_failing := $(shell cat tests/failing.symbolic)
+symbolic_tests_passing := $(filter-out $(symbolic_tests_failing), $(symbolic_tests))
+
+test-symbolic:         $(symbolic_tests_passing:=.symbolic)
+test-symbolic-failing: $(symbolic_tests_failing:=.symbolic)
+
+EXPECTED_EXITCODE = 0
+tests/symbolic/%.stuck.tzt.symbolic: EXPECTED_EXITCODE = 127
+tests/symbolic/%.fail.tzt.symbolic:  EXPECTED_EXITCODE = 1
```

Comment:
@ehildenb is this a reasonable way of doing things?

Reply:
I guess I would do it differently (and avoid altogether the need for `check-exit-code` script).

Specifically, I would pass in the desired result directly to `michelson-test-check` directly, eg.:

```
$(TEST) run --backend symbolic $< --search-final --output json \
  | $(LIB_DIR)/michelson-test-check $(dir $(symbolic_kompiled)) $(EXPECTED_EXITCODE) > /dev/null
```

Reply:
My thought in keeping it separate was that users may want to use `michelson-test-check` outside of the test framework, to quickly inspect the very fairly output generated but `--search-final`

Reply:
Not sure what "very fairly output" means.

But anyway, if you have functionality that you think users would want often, I would (i) first gather evidence that it's wanted, and (ii) then build it into the `kmich` script with a nice wrapper and documentation instead of having it be a separate script.

So perhaps you should consider either a `./kmich symbolic` or a `./kmich search` or a `./kmich symbtest` which wraps up all this functionality, so that users can access it directly.

But note, the `kmich` script should be for _end-user_ functionality, _not_ for testing harnesses. So if this is only useful as a testing harness thing, then I wouldn't put it there.

Reply:
The `michelson-test-check` binary can also be thought of as part of our "product", since we are shipping a testing framework. Whereas `check-exit-code` is used to test this test framework.

Reply:
I had meant "fairly large output".

> So perhaps you should consider either a ./kmich symbolic or a ./kmich search or a ./kmich symbtest which wraps up all this functionality, so that users can access it directly.

Since parsing k output in bash is tricky, we'd need `kmich` to wrap this script. OK if I do that in a future pull request?

> But note, the kmich script should be for end-user functionality, not for testing harnesses. So if this is only useful as a testing harness thing, then I wouldn't put it there.

Our product is a symbolic testing framework. This is end-user functionality in that context.